tools/jit/gen_unboxing.py

```diff
@@ -0,0 +1,173 @@
+                unboxing.generate_unboxed_kernel_call(f, sig, arguments)
+            )
+
+            return f"""
+// aten::{f.func}
+TORCH_API void {f.func.name.unambiguous_name()}(Stack & stack) {{
+    {code}
+    drop(stack, {len(args)});
+    {func_call_and_push}
+}}
+"""
+
+
+@dataclass(frozen=True)
+class ComputeUnboxingWrapper:
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> str:
+        # We unconditionally generate function wrappers,
+        sig_group = CppSignatureGroup.from_native_function(
+            f, method=False, fallback_binding=f.manual_cpp_binding
+        )
+
+        sig = sig_group.signature
+
+        # escape double quote in schema, get rid of extra double quotes
+        schema = json.dumps(sig.func.__str__())[1:-1]
+
+        return f"""
+OperatorGenerator(
+    TORCH_SELECTIVE_SCHEMA("aten::{schema}"),
+    [](Stack & stack) {{
+        RECORD_FUNCTION("{sig.name()}", std::vector<c10::IValue>());
+        at::unboxing::{f.func.name.unambiguous_name()}(stack);
+    }},
+    aliasAnalysisFromSchema()
+),
+"""
+
+
+def gen_unboxing(
+        *,
+        native_functions: Sequence[NativeFunction],
+        cpu_fm: FileManager,
+) -> None:
+    def key_func(fn: Union[NativeFunction, NativeFunctionsGroup]) -> str:
+        return fn.root_name
+
+    cpu_fm.write_sharded(
+        "CodegenFunctions.cpp",
+        native_functions,
+        key_fn=key_func,
+        env_callable=lambda fn: {
+            "definitions": [ComputeUnboxingFunctions(Target.DEFINITION)(fn)]
+        },
+        num_shards=5,
+        sharded_keys={"definitions"},
+    )
+    cpu_fm.write(
+        "CodegenFunctions.h",
+        lambda: {
+            "declarations": list(
+                mapMaybe(ComputeUnboxingFunctions(Target.DECLARATION), native_functions)
+            ),
+        },
+    )
+    cpu_fm.write_sharded(
+        "CodegenUnboxingWrappers.cpp",
+        native_functions,
+        key_fn=key_func,
+        env_callable=lambda fn: {"unboxed_ops": [ComputeUnboxingWrapper()(fn)]},
+        num_shards=5,
+        sharded_keys={"unboxed_ops"},
+    )
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Generate unboxing source files")
+    parser.add_argument(
+        "-s",
+        "--source-path",
+        help="path to source directory for ATen",
+        default="aten/src/ATen",
+    )
+    parser.add_argument(
+        "-d", "--install_dir", help="output directory", default="build/aten/src/ATen"
+    )
+
+    options = parser.parse_args()
+
+    native_yaml_path = os.path.join(options.source_path, "native/native_functions.yaml")
+    parsed_yaml = parse_native_yaml(native_yaml_path)
+    native_functions, backend_indices = (
+        parsed_yaml.native_functions,
+        parsed_yaml.backend_indices,
+    )
+
+    template_dir = os.path.join(options.source_path, "templates")
+
+    def make_file_manager(install_dir: str) -> FileManager:
```

Comment:
nit: move this into `tools/codegen/utils.py` so we can re-use it between here and `gen.py`